### DISCUSSION TITLE
Verify tls version used from Client side

### DISCUSSION THREAD

**Discussion by som-nitjsr** [dotnet/dotnet-docker#2589] [Q&A]
From aspnet core 3.1 Linux container hosted in aks I am calling azure sql server and azure key vault. can u tell me how I can prove my client is using tls1.2 while on azure sql server tls 1.0 1.1 and 1.2 in enabled. On Azure Sql server i have not enabled minimum tls version.
Azure Key vault does not support minumum tls version..
I am using entity framework core 3.1.
i want verification from client side.
I need this for my audit.

I am able to see in **/etc/ssl/openssl.config** 

[ssl_sect]
system_default = system_default_sect

[system_default_sect]
MinProtocol = TLSv1.2
CipherString = DEFAULT@SECLEVEL=2

Is this enough to say that my container is using tls 1.2 to connect azure sql server and azure key vault.

**Comment by MichaelSimons** [comment]
I would recommend inspecting the outgoing request via a tool like [fiddler](https://www.telerik.com/fiddler) or .NET tracing.

[Stack Overflow question on this topic](https://stackoverflow.com/questions/34276358/how-to-test-which-version-of-tls-my-net-client-is-using)
[Using dotnet-monitor to gather traces](https://devblogs.microsoft.com/dotnet/introducing-dotnet-monitor/)

**Reply by som-nitjsr** [reply]
I have tested and it is using tls 1.2.
I have referred this blog. 
is this sentence from below blog is documented some where? 
**First, the difference in the containers between 2.1 and 3.1 is that support for TLS 1.0, 1.1 was dropped. Subsequently, it can only communicate on TLS 1.2 now.** 
http://blog.travisgosselin.com/tls-1-0-1-1-docker-container-support/
